functorch/experimental/_map.py

```diff
@@ -65,10 +84,10 @@ def create_fw_bw_graph(f, num_mapped_args, *args):
     # when creating the output node, it fails to associate the wrapped tensor with its proxy.
     # Instead, it will create _tensor_constant as output.
 
-    with suspend_functionalization():
+    with suspend_functionalization(), unset_functional_temporarily():
```

Comment:
IMO FunctionalTensorMode should automatically `unset_functional_temporarily` to be analogous to functorch.functionalize. Would there be problems if we did that?

Reply:
Hmm I don't think I follow - what behavior in `functorch.functionalize` is that analogous to?

(also - I'm probably missing something, but I'm not sure how that's related to this line. If I understand correctly, we want to ahead-of-time trace out a fw and bw graph here, which we will then invoke from the autograd rule, and then functionalize later, which is why we're ok with turning off functionalization now. But doesn't that mean that we need to explicitly disable functionalization here? (both the C++ variant by disabling the `Functionalize` dispatch key, and the python variant by temporarily disabling any ambient `FunctionalTensorMode` that might be active).

Reply:
sorry, I misread unset_functional_temporarily() as "turn on the functionalization key", we are actually disabling it to be in-line with suspend_functionalization() so that makes sense to me